lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -336,8 +342,15 @@ bool TempRValueOptPass::tryOptimizeCopyIntoTemp(CopyAddrInst *copyInst) {
   if (!tempObj)
     return false;
 
-  assert(tempObj != copyInst->getSrc() &&
-         "can't initialize temporary with itself");
+  // The copy's source address must not be a scoped instruction. When the
+  // temporary object is eliminated, it's uses are replaced with the copy's
+  // source. Therefore, the source address must be valid at least until the next
+  // instruction that may write to or destroy the source. End-of-scope markers
+  // do not write to or destroy memory, so scoped addresses are not valid
+  // replacements.
```

Comment:
Can you provide an example here or define what is a "scoped" instruction. It is undefined when I pretend to not know what you probably mean.

Reply:
Yes, good idea.